ft/jetton-minter-discoverable.fc

```diff
@@ -0,0 +1,150 @@
+;; Jettons discoverable smart contract
```

Comment:
Why would people who deploy jettons choose this variant over other variants? Is there a downside to supporting discoverability in all minter variants?

I believe that in order to support future dex inclusion, the jetton-minter (and jetton-minter ICO) variants should support discoverability instead of it being a separate variant to prevent confusion. 

Also consider maintainability - if there are future changes to the standard, it would have to be done across all variants, which is prone to errors.

Reply:
as I understand, 
future jettons contracts should include *get_wallet_address* functionality,

jettons with upgradeable metdata should deploy a discovery contract and update the metadata with discover address 

**jettons with non upgradble metadata** -  for example a dex that want to support this kind of jettons, has to to pass the discover-jetton address in the init message  from the dapp ( client ). 
 its not very secure but possible.